File: backend/member-service/src/main/java/com/example/memberservice/member/dto/ResponseMemberDto.java
package com.example.memberservice.member.dto;

import com.example.memberservice.member.entity.Role;
import com.example.memberservice.profile.dto.ProfileDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResponseMemberDto implements Serializable {

    private Long id;
    private String userId;
    private String auth;
    private Role role;
    private String email;
    private String username;
    private String statusMessage;
    private String profileImage;
    private String wallpaperImage;
    private List<ProfileDto> profiles;

    public ResponseMemberDto(MemberDto memberDto, List<ProfileDto> profiles) {
        this.id = memberDto.getId();
        this.userId = memberDto.getUserId();
        this.auth = memberDto.getAuth();
        this.role = memberDto.getRole();
        this.email = memberDto.getEmail();
        this.username = memberDto.getUsername();
        this.statusMessage = memberDto.getStatusMessage();
        this.profileImage = memberDto.getProfileImage();
        this.wallpaperImage = memberDto.getWallpaperImage();
        this.profiles = profiles;
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/member/dto/ResponseMemberDto.kt
package com.example.memberservice.member.dto

import com.example.memberservice.member.entity.Role
import com.example.memberservice.profile.dto.ProfileDto
import java.io.Serializable

data class ResponseMemberDto(
    var id: Long? = null,
    var userId: String? = null,
    var auth: String? = null,
    var role: Role? = null,
    var email: String? = null,
    var username: String? = null,
    var statusMessage: String? = null,
    var profileImage: String? = null,
    var wallpaperImage: String? = null,
    var profiles: List<ProfileDto>? = null,
) : Serializable {

    constructor(memberDto: MemberDto, profiles: List<ProfileDto>?) : this(
        id = memberDto.id,
        userId = memberDto.userId,
        auth = memberDto.auth,
        role = memberDto.role,
        email = memberDto.email,
        username = memberDto.username,
        statusMessage = memberDto.statusMessage,
        profileImage = memberDto.profileImage,
        wallpaperImage = memberDto.wallpaperImage,
        profiles = profiles,
    )

    companion object {
        @JvmStatic
        fun from(memberDto: MemberDto, profiles: List<ProfileDto>?): ResponseMemberDto = ResponseMemberDto(memberDto, profiles)
    }
}
